src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1398,6 +1398,7 @@ internal bool IsNonAttributedType
             }
 
             [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
+            [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods, typeof(KeyValue<,>))]
```

Comment:
Would it be better to use the new `GetMemberWithSameMetadataDefinitionAs` instead? We should be able to avoid a DynamicDependency here with it.

See #53704 for examples on how to use it.